File: Definitions/Game_Definitions/classes/ModifyAttackCritChanceEffector.lua
---@meta
---@diagnostic disable

---@class ModifyAttackCritChanceEffector : ModifyAttackEffector
---@field applicationChanceMods gamedataStatModifier_Record[]
ModifyAttackCritChanceEffector = {}

---@return ModifyAttackCritChanceEffector
function ModifyAttackCritChanceEffector.new() return end

---@param props table
---@return ModifyAttackCritChanceEffector
function ModifyAttackCritChanceEffector.new(props) return end

---@param record TweakDBID|string
---@param parentRecord TweakDBID|string
function ModifyAttackCritChanceEffector:Initialize(record, parentRecord) return end

---@param owner gameObject
function ModifyAttackCritChanceEffector:RepeatedAction(owner) return end

function ModifyAttackCritChanceEffector:Uninitialize() return end

File: DecompiledGameScripts/core/gameplay/effectors/damage/modifyAttackCritChanceEffector.swift

public class ModifyAttackCritChanceEffector extends ModifyAttackEffector {

  public let m_applicationChanceMods: [wref<StatModifier_Record>];

  protected func Initialize(record: TweakDBID, game: GameInstance, parentRecord: TweakDBID) -> Void {
    TweakDBInterface.GetModifyAttackCritChanceEffectorRecord(record).CritChance(this.m_applicationChanceMods);
  }

  protected func Uninitialize(game: GameInstance) -> Void;

  protected func RepeatedAction(owner: ref<GameObject>) -> Void {
    let critChance: Float;
    let hitEvent: ref<gameHitEvent> = this.GetHitEvent();
    if !IsDefined(hitEvent) {
      return;
    };
    critChance = RPGManager.CalculateStatModifiers(this.m_applicationChanceMods, owner.GetGame(), owner, Cast<StatsObjectID>(owner.GetEntityID()));
    hitEvent.attackData.SetAdditionalCritChance(critChance);
  }
}
